_posts/2023-06-30-trino-fest-2023-apacheiceberg.md

```diff
@@ -0,0 +1,61 @@
+---
+layout: post
+title: "CDC patterns in Apache Iceberg"
+author: "Ryan Blue, Ryan Duan"
+excerpt_separator: <!--more-->
+image: /assets/blog/trino-fest-2023/ApacheIceberg.png
+show_pagenav: false
+---
+
+Have you ever wanted to keep your data in a table and have an efficient way to
+interact with them? Iceberg, an open standard table format, is
+exactly what you need. What makes Iceberg unique is its use of change data
+capture (CDC). Co-creator of Apache Iceberg, Ryan Blue, gave a talk at [Trino Fest 2023]({% post_url
+2023-06-20-trino-fest-2023-recap %}) this past week detailing the trade-offs
+between each pattern that can be used for writing CDC streams into Iceberg
+tables.
+
+<!--more-->
+
+{% youtube GM7EvRc7_is %}
+
+<a class="btn btn-pink btn-md" target="_blank" href="/assets/blog/trino-fest-2023/TrinoFest2023Iceberg.pdf">
+  Check out the slides!
+</a>
+
+## Recap
+
+To begin, what is CDC and why should you use it? CDC is the idea that relational
+or transactional tables are modified, you emit an update stream to keep copies
+in sync so that you capture changes to tables as they happen. As Ryan states,
+"[CDC] is very lightweight on the source database ... rather than being super
+careful with what we run on the database, what we want to do is just make a
+copy of it very easily and maintain that copy." Ryan continues giving an
+example of a bank using a transactional table in Iceberg to offer some
+context on what's going on.
+
+With this knowledge, you can see some problems that makes CDC difficult:
+
+* Lower latency means more work
+* Write amplification - the work necessary to balance the trade-off between
+  efficiency at write time and efficiency at read time
+* Batch writes with double update and possible inconsistency
+* Read requirements with the different types of deletes in a table
+
+With these types of problems, the importance of the trade-offs between the
+different patterns rises due to the need for utmost efficiency. The first
+trade-off that Ryan talks about is the storage trade-offs between using direct
```

Comment:
Need to decide whether this is singular or plural. "The first trade-off... is the trade-offs" doesn't work.